### PR TITLE
User Profiles 

### PR DESCRIPTION
**Pull request by tuckermullen**
Had some issues with writing the show action in the controller but I was able to research and figure it out this morning. I'm still having some issues writing tests for this project. I don't know if it's just me over thinking it but I can't seem to figure out what needs to be done versus what doesn't need to be done with Devise doing so much for us. I have two tests total for this project and one is failing. Overall, this code does in fact check the boxes when it comes to what the checkpoint asks me. It does redirect to the show page when you sign in and it does have a profile page that shows something. I'm not sure what exact details the checkpoint wants us to show since the only real user detail is an email address. @bjblock

### CODE REVIEW

File: spec/controllers/users_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+require 'rails_helper'
+
+RSpec.describe UsersController, type: :controller do
+
+  describe "GET #show" do
+    it "returns http success" do
+      get :show
+      expect(response).to have_http_status(:success)
```
Comment (bjblock) [inline spec/controllers/users_controller_spec.rb]: I'm guessing this is the test that's failing?

Comment (tuckermullen) [inline spec/controllers/users_controller_spec.rb]: Yes, it's giving me a 302 response which I know is a redirect. I'm all
ready testing that in the test below that one so I obviously don't need to
be testing the same thing twice right?
On Tue, Jan 12, 2016 at 12:25 PM bjblock notifications@github.com wrote:

> In spec/controllers/users_controller_spec.rb
> https://github.com/tuckermullen/blocitoff/pull/2#discussion_r49508268:
> 
> > @@ -0,0 +1,16 @@
> > +require 'rails_helper'
> > +
> > +RSpec.describe UsersController, type: :controller do
> > +
> > -  describe "GET #show" do
> > -    it "returns http success" do
> > -      get :show
> > -      expect(response).to have_http_status(:success)
> 
> I'm guessing this is the test that's failing?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/tuckermullen/blocitoff/pull/2/files#r49508268.
> 
> ## 

_Tucker Mullen_
253-208-9095

Comment (bjblock) [inline spec/controllers/users_controller_spec.rb]: remember, the trickiest part of testing is setting up your data correctly and making sure you're testing the right "thing".  besides the expected response, what is the biggest difference between these 2 tests?

**Comment by tuckermullen** [comment]
The first test makes sure that the show view actually returns a successful http status where as the other renders the actual show view and not something else. Would you use an expected response like this instead?:

```
it "renders the #show view" do
   get :show, {id: @user.id}
   expect(response).to render_template :show
end
```

**Comment by tuckermullen** [comment]
Also, the first test is calling `get: show` where the second is calling the same with `{id: @user.id}` which I used to make sure that the show view was going to the correct users profile show view. Is this not enough? I assumed it would be since it's routed this way in the routes file.

**Pull request merged**